hawkular-dmr-client/src/main/java/org/hawkular/dmr/api/OperationBuilder.java

```diff
@@ -0,0 +1,751 @@
+/*
+ * Copyright 2015 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.hawkular.dmr.api;
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.StringTokenizer;
+import java.util.stream.Collectors;
+
+import org.jboss.as.controller.client.ModelControllerClient;
+import org.jboss.as.controller.descriptions.ModelDescriptionConstants;
+import org.jboss.dmr.ModelNode;
+import org.jboss.dmr.Property;
+
+/**
+ * A set of type safe builders for building and executing DMR operations.
+ * <p>
+ * Example usage:
+ *
+ * <pre>
+ * &#47;&#47; read a datasource
+ * ModelControllerClient mcc = ...;
+ * ModelNode myDs = OperationBuilder.readResource()
+ *          .address().subsystemDatasources().segment("data-source", "myDatasource").parentBuilder()
+ *          .includeDefaults()
+ *          .recursive()
+ *          .execute(mcc)
+ *          .assertSuccess()
+ *          .getResultNode();
+ *
+ * &#47;&#47; remove a datasource with another way of providing an address
+ * OperationBuilder.remove()
+ *          .address().segments("/subsystem-datasources/data-source=myDatasource").parentBuilder()
+ *          .execute(mcc)
+ *          .assertSuccess();
+ * </pre>
+ *
+ * @author <a href="https://github.com/ppalaga">Peter Palaga</a>
+ */
+public class OperationBuilder implements SubsystemDatasourceConstants, SubsystemLoggingConstants {
```

Comment:
> Is this only for datasources?

No, it is general purpose. `implements` is just "misused" to import all constants from the interface. 

If your objection is that a general-purpose builder should not rely on/use subsystem specific constants, then... well... that's a good question :) I think it is OK to have names of subsystems here, because the set of subsystems is fairly stable. And having it here, we can offer handy shortcuts like `AddressBuilder.subsystemDatasources()` https://github.com/hawkular/hawkular-agent/pull/76/files#diff-650a4b8ae9cc61f2557a8bed543e3087R238